src/System.Private.CoreLib/src/System/Diagnostics/Eventing/EventPipePayloadDecoder.cs

```diff
@@ -110,16 +110,29 @@ internal static object[] DecodePayload(ref EventSource.EventMetadata metadata, R
                 }
                 else if (parameterType == typeof(string))
                 {
-                    ReadOnlySpan<char> charPayload = MemoryMarshal.Cast<byte, char>(payload);
-                    int charCount = charPayload.IndexOf('\0');
-                    if (charCount < 0)
+                    // Try to find null terminator (0x00) from the byte span
+                    // NOTE: we do this by hand instead of using payload.IndexOf('\n') because payload may be unaligned due to 
+                    // mixture of different types being stored in the same buffer. (see eventpipe.cpp:CopyData)
+                    int byteCount = -1;
+                    for (int j = 1; j < payload.Length; j+=2)
                     {
+                        if (payload[j-1] == (byte)(0) && payload[j] == (byte)(0))
+                        {
+                            byteCount = j;
```

Comment:
Nit: its a little odd to compute this as the 'byteCount'. For example in the string cat, I'd expect byteCount = 8 (c,a,t,0 * sizeof(char)) but this compute 7. The code below handles it correctly so no bug, just an unexpected convention.